trusted_application_demo/ta_security_api/src/ta_mainpoint.c

```diff
@@ -0,0 +1,341 @@
+//
+// Copyright (c) 2018, ARM Limited.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+
+#include "./user_ta_header_defines.h"
+#include "ta_security_api/ta_public.h"
+
+#include "ta_security_api/ta_aes.h"
+#include "ta_security_api/ta_rsa.h"
+#include "ta_security_api/ta_sha.h"
+
+/*
+ * Called when the instance of the TA is created. This is the first call in
+ * the TA.
+ */
+TEE_Result
+TA_CreateEntryPoint(void)
+{
+  return TEE_SUCCESS;
+}
+
+/*
+ * Called when the instance of the TA is destroyed if the TA has not
+ * crashed or panicked. Last call in the TA.
+ */
+void TA_DestroyEntryPoint(void) {}
+
+/*
+ * Called when a new session is opened to the TA.
+ */
+TEE_Result
+TA_OpenSessionEntryPoint(
+  uint32_t param_types,
+  TEE_Param params[4],
+  void ** sessionContext)
+{
+  TEE_Result result;
+  uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE,
+      TEE_PARAM_TYPE_NONE,
+      TEE_PARAM_TYPE_NONE,
+      TEE_PARAM_TYPE_NONE);
+  if (param_types != exp_param_types) {
+    return TEE_ERROR_BAD_PARAMETERS;
+  }
+
+  result = ta_initialize_digest();
+  if (TEE_SUCCESS != result) {
+    return result;
+  }
+
+  result = ta_rsa_prepare_key();
+  if (TEE_SUCCESS != result) {
+    return result;
+  }
+
+  (void)params;
+  (void)sessionContext;
```

Comment:
is the goal that this function will be changed later to allow user to pass parameters and context? Maybe a TODO explaining what the next steps are would be helpful (same below for TA_CloseSessionEntryPoint)

Reply:
Optee-OS' Makefile treats unused arguments as error. I am not sure how the application could use them, even later on :D